### PR TITLE
Setup pipeline-as-code

### PR DESCRIPTION
**Pull request by lcarva**
https://issues.redhat.com/browse/HACBS-1494

Signed-off-by: Luiz Carvalho <lucarval@redhat.com>

### CODE REVIEW

**Comment by lcarva** [comment]
Cool! I see the "Pipelines as Code CI" check on this PR already! It is currently marked as skipped with the following message:
```
cannot find a namespace match for https://github.com/hacbs-contract/golden-container
```

This should fix that: https://github.com/redhat-appstudio/infra-deployments/pull/1048

(I think that's all that is needed.)

**Comment by Michkov** [comment]
I also added permission to robot account to write to the redhat-appstudio/ec-golden-image

/retest

File: .tekton/pull-request.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - name: output-image
+      value: quay.io/redhat-appstudio/pull-request-builds:ec-golden-image-{{revision}}
+    - name: dockerfile
+      value: ./Containerfile
+    - name: path-context
+      value: .
+  pipelineRef:
+    name: docker-build
+    # TODO: Use replace git-commit tag with a floating tag so it can be managed by renovate bot.
+    bundle: quay.io/redhat-appstudio-tekton-catalog/pipeline-docker-build:165e81585fb7afd4fb17be0ef3851cb3962e281d@sha256:21e4124faa37485c8c170f3b367b4ae9334c1b6bc5420b802fbdcebd31069583
```
Comment (Michkov) [inline .tekton/pull-request.yaml:27]: this bundle has currently set hacbs=false, so you can enable it via param. This behavior will change soon probably.

Comment (lcarva) [reply .tekton/pull-request.yaml:29]: Thanks!

**Review by Michkov — APPROVED** [review]

**Comment by lcarva** [comment]
Build is working!

However, Chains is throwing unauthorized errors. The theory is that the mounted secret on the Chains deployment is getting used which does not have access to push content to the repo. The attempted fix is to make the mounted secret more specific since it's only meant for one specific repository. This should cause Chains to ignore it in this case and use the secret from the service account instead. We'll try that tomorrow. :pray:

**Comment by Michkov** [comment]
Secret updated but still not working. Re-created tekton-chains pod to get new secret and testing again:

/retest

**Comment by Michkov** [comment]
Still not working failing on `POST https://quay.io/v2/redhat-appstudio/pull-request-builds/blobs/uploads/: UNAUTHORIZED: access to the requested resource is not authorized`. The shared-secret has been updated - PLNSRVCE-941

**Comment by lcarva** [comment]
The issue appears to be caused by [google/go-containerregistry#1510](https://www.github.com/google/go-containerregistry/issues/1510)

**Comment by lcarva** [comment]
The build pipeline is working here. What is failing is Chains being able to sign the image due to the issue mentioned above. I can't think of a workaround with this change in place, [google/go-containerregistry#1512](https://www.github.com/google/go-containerregistry/pull/1512).

The newer version of Chains does bring a newer version of the `go-containerregistry` library that changes the precedence order. The auth key is first searched in the k8s secrets rather than on the mounted docker config. This should also fix the issue with Chains signing, but may actually push the problem somewhere else.

I think there's benefit in merging this PR since the issue we're seeing is not related to it, and won't be fixed by it. Also, the issue impacts other repos as well, not just this one.

@zregvart , wdyt about merging this, and following up on resolving the Chains issue separately?

**Review by zregvart — APPROVED** [review]

**Pull request merged**